### DISCUSSION TITLE
jetcd: troubles migrating to etcd 3.5.9 from 3.5.4

### DISCUSSION THREAD

**Discussion by lburgazzoli** [etcd-io/etcd#15969] [General]
Hello,

I'm the maintainer of the [jetcd project ](https://github.com/etcd-io/jetcd) and while trying to upgrade the image used for integration test from etcd `3.5.4` to `3.5.9`, I noticed that a tests that uses jwt as auth method, has started to fail with `etcdserver: invalid auth management` but nothing has been changed on the library side and there's not so much info in the etcd output to help troubleshooting the issue.

```
--auth-token jwt,pub-key=/etc/ssl/etcd/server.pem,priv-key=/etc/ssl/etcd/server-key.pem,sign-method=RS256,ttl=1s
```

**Comment by ahrtr** [comment]
@lburgazzoli would you mind to provide more detailed info? e.g. test case name, and more log.

cc @mitake

**Comment by lburgazzoli** [comment]
An example of the running action failing is here: https://github.com/etcd-io/jetcd/pull/1145 and the failing test is this one https://github.com/etcd-io/jetcd/blob/main/jetcd-core/src/test/java/io/etcd/jetcd/impl/WatchTokenExpireTest.java

The initial investigation shows that the failure happens here: https://github.com/etcd-io/jetcd/blob/73749064e80b196b54e217e17eaac1d07d5f53b3/jetcd-core/src/test/java/io/etcd/jetcd/impl/WatchTokenExpireTest.java#L75-L76 but I can't figure out much as what I see in the etcd logs is nothing more that `etcdserver: invalid auth management`

**Reply by mitake** [reply]
@lburgazzoli the line tries to grant an empty range `[key, key)`, recently we fixed etcd for disallowing to create such empty permissions (actually they are simply ignored in the old versions). The role in the test is used for root role, so it can access to any keys. Could you drop the line of `roleGrantPermission()` or grant a permission for a single key?

**Reply by lburgazzoli** [reply]
ok, now it does not fail anymore for auth related issues but for another one I must digg into, thx for the support
